### ISSUE TITLE
ScrimInsetsFrameLayout with shadows

### ISSUE THREAD

**Issue by ZieIony** [ZieIony/Carbon#56]
Copied from #54 

[ScrimInsetsFrameLayout](https://github.com/google/iosched/blob/master/android/src/main/java/com/google/samples/apps/iosched/ui/widget/ScrimInsetsFrameLayout.java), a layout Google made to enable the drawer to be tinted underneath the status bar (as in the guidelines), stops working when it's enclosed in a Carbon FrameLayout (which is the only way I could get the shadow to work). It stops short below the status bar. I tried making it extend Carbon's FrameLayout instead of Android's, but it didn't change anything.
My current layout:

```
<carbon.widget.FrameLayout
    android:layout_width="280dp"
    android:layout_height="match_parent"
    android:layout_gravity="start"
    android:fitsSystemWindows="true">

    <mypackage.ScrimInsetsFrameLayout
        android:id="@+id/scrimInsetsFrameLayout"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:background="@android:color/white"
        android:fitsSystemWindows="true"
        app:carbon_elevation="4dp"
        app:insetForeground="#4000">

        <fragment
            android:id="@+id/fragment_drawer"
            android:name="mypackage.DrawerFragment"
            android:layout_width="match_parent"
            android:layout_height="match_parent" />
    </mypackage.ScrimInsetsFrameLayout>
</carbon.widget.FrameLayout>
```

Ideally, it would just be a `carbon.widget.ScrimInsetsFrameLayout` containing the fragment.
Since ScrimInsetsFrameLayout is practically required to implement a Material-style drawer, do you think you could make a Carbon widget of that too?

**Comment by ZieIony** [comment]
I've just released version 0.7.1 with a drawer sample. I prefer using StatusBar and NavigationBar views for drawing these parts of UI. You may find this file interesting: https://github.com/ZieIony/Carbon/blob/master/samples/src/main/res/layout/activity_drawer.xml

ScrimInsetsLayout is there as well. I had to push some important bugfixes asap, so there's no sample for that layout yet. Maybe next week :)

**Comment by ZieIony** [comment]
ScrimInsetsLayout seems to work, so I'm closing this issue. Any potential bugs will be handled in separate issues
